processor/src/main/resources/org.mapstruct.ap.model.IterableMappingMethod.ftl

```diff
@@ -50,6 +56,21 @@
         <#if !existingInstanceMapping>
             <@includeModel object=resultElementType/>[] ${resultName} = new <@includeModel object=resultElementType/>[<@iterableSize/>];
         </#if>
+    <#else>
+        <#if existingInstanceMapping>
```

Comment:
I've moved those parts together so that there is only one block of result initialization so that I can follow with one single block for the ```beforeMappingReferencesWithMappingTarget``` references (that are ```@BeforeMapping``` methods with an ```@MappingTarget``` parameter and so they need an initialized target value.